### PR TITLE
added ShoppingCart functionality

### PR DESCRIPTION
**Pull request by citruscaviar**

### CODE REVIEW

File: src/main/java/com/onlinebookstore/store/service/impl/ShoppingCartServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    @Transactional
+    public ShoppingCartDto removeCartItem(Long userId, Long cartItemId) {
+        ShoppingCart shoppingCart = getShoppingCartByUserId(userId);
+        CartItem cartItem = cartItemRepository.findByIdAndShoppingCartId(
+                cartItemId, shoppingCart.getId());
+        shoppingCart.removeItemFromCart(cartItem);
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    public void createNewShoppingCart(User user) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/store/service/impl/ShoppingCartServiceImpl.java:75]:
```suggestion
    @Override
    public void createNewShoppingCart(User user) {
```

File: src/main/java/com/onlinebookstore/store/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -22,6 +23,7 @@ public class UserServiceImpl implements UserService {
     private final RoleRepository roleRepository;
     private final PasswordEncoder passwordEncoder;
     private final UserMapper userMapper;
+    private final ShoppingCartService shoppingCartService;
 
     @Override
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/store/service/impl/UserServiceImpl.java:28]:
```suggestion
    @Override
    @Transactional
```

File: src/main/java/com/onlinebookstore/store/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -36,6 +38,7 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto) {
                 .orElseThrow(
                     () -> new EntityNotFoundException("Can't find role by name"));
         user.setRoles(Set.of(role));
+        shoppingCartService.createNewShoppingCart(userRepository.save(user));
         return userMapper.toDto(userRepository.save(user));
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/store/service/impl/UserServiceImpl.java:42]:
```suggestion
        userRepository.save(user);
        shoppingCartService.createNewShoppingCart(user);
        return userMapper.toDto(user);
```

File: src/main/resources/db/changelog/changes/09-create-cart_item-table.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+                  name: shopping_cart_id
+                  type: bigint
+              - column:
+                  name: book_id
+                  type: bigint
+              - column:
+                  name: quantity
+                  type: int
+              - column:
+                  name: is_deleted
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/09-create-cart_item-table.yaml:26]: didn't find soft delete approach in CartItem entity

File: src/main/resources/db/changelog/changes/08-create-shopping_cart-table.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: user_id
+                  type: bigint
+                  constraints:
+                    nullable: false
+              - column:
+                  name: is_deleted
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/08-create-shopping_cart-table.yaml:22]: didn't find soft delete approach in ShoppingCart entity

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/com/onlinebookstore/store/repository/CartItemRepository.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    @Modifying
+    @Query("UPDATE CartItem item "
+            + "SET item.quantity = :quantity "
+            + "WHERE item.id = :id")
+    void updateCartItemQuantityById(int quantity, Long id);
+
+    @Query("select c from CartItem c where c.id = ?1 and c.shoppingCart.id = ?2")
+    CartItem findByIdAndShoppingCartId(Long itemId, Long cartId);
+}
+
```
Comment (okuzan) [inline src/main/java/com/onlinebookstore/store/repository/CartItemRepository.java:18]:
```suggestion
```

**Review by okuzan — APPROVED** [review]

**Pull request merged**